app/src/main/kotlin/at/bitfire/davdroid/sync/SyncNotificationManager.kt

```diff
@@ -0,0 +1,283 @@
+        } else {
+            channel = notificationRegistry.CHANNEL_SYNC_ERRORS
+            priority = NotificationCompat.PRIORITY_DEFAULT
+        }
+
+        val builder = NotificationCompat.Builder(context, channel)
+        builder.setSmallIcon(R.drawable.ic_sync_problem_notify)
+            .setContentTitle(localCollection.title)
+            .setContentText(message)
+            .setStyle(NotificationCompat.BigTextStyle(builder).bigText(message))
+            .setSubText(account.name)
+            .setOnlyAlertOnce(true)
+            .setContentIntent(
+                TaskStackBuilder.create(context)
+                    .addNextIntentWithParentStack(contentIntent)
+                    .getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
+            )
+            .setPriority(priority)
+            .setCategory(NotificationCompat.CATEGORY_ERROR)
+        viewItemAction?.let { builder.addAction(it) }
+
+        builder.build()
+    }
+
+    /**
+     * Sends a notification to inform the user that a push notification has been received, the
+     * sync has been scheduled, but it still has not run.
+     *
+     * @param notificationTag   The tag to use for the notification.
+     * @param collection        The affected collection.
+     * @param fileName          The name of the file containing the invalid resource.
+     * @param title             The title of the notification.
+     */
+    fun notifyInvalidResource(
+        authority: String,
+        notificationTag: String,
+        collection: Collection,
+        e: Throwable,
+        fileName: String,
+        title: String
+    ) {
+        notificationRegistry.notifyIfPossible(NotificationRegistry.NOTIFY_INVALID_RESOURCE, tag = notificationTag) {
+            val intent = buildDebugInfoIntentForLocalResource(authority, e, null, collection.url.resolve(fileName))
+
+            val builder = NotificationCompat.Builder(context, notificationRegistry.CHANNEL_SYNC_WARNINGS)
+            builder.setSmallIcon(R.drawable.ic_warning_notify)
+                .setContentTitle(title)
+                .setContentText(context.getString(R.string.sync_invalid_resources_ignoring))
+                .setSubText(account.name)
+                .setContentIntent(
+                    TaskStackBuilder.create(context)
+                        .addNextIntent(intent)
+                        .getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
+                )
+                .setAutoCancel(true)
+                .setOnlyAlertOnce(true)
+                .priority = NotificationCompat.PRIORITY_LOW
+            builder.build()
+        }
+    }
+
+    /**
+     * Dismisses the (error) notification for a specific collection.
+     *
+     * @param localCollectionTag The tag of the local collection which is used as notification tag also.
+     */
+    fun dismissCollectionSpecificNotification(localCollectionTag: String) =
```

Comment:
Can we shorten the method name a bit? It's always a challenge to say everything we want to say, but on the other side keep the method name short.

Maybe `dismissCollectionNotification` as a trade-off?